### PR TITLE
[ContainerRegistry] update for better codegen for data plane Track 2 SDK libraries

### PR DESCRIPTION
**Pull request by jeremymeng**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [x] Other, please clarify: **update swagger to allow better codegen for data plane Track 2 SDK libraries**
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [x] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by annelo-msft** [comment]
@jeremymeng , if the Track 1 swagger filepath is  `Microsoft.ContainerRegistry/preview/2019-08-15/`, would it make sense to move the Track 2 swagger to a filepath with a different date?  Or does the date mean a specific version of the service, and so the Track 2 file describes the same one and we should use the same date?

**Comment by jeremymeng** [comment]
> Or does the date mean a specific version of the service

yes I believe it is the service version. At least that what I've been seeing in Storage.

File: specification/containerregistry/data-plane/Azure.ContainerRegistry/readme.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+---
+# Code Generation
+
+## C#
+
+These settings apply only when `--csharp` is specified on the command line.
+
+``` yaml $(csharp)
+csharp: true
```
Comment (jeremymeng) [inline specification/containerregistry/data-plane/Azure.ContainerRegistry/readme.md:37]: @annelo-msft please review

File: specification/containerregistry/data-plane/Azure.ContainerRegistry/readme.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    transform: >
+      $["x-accessibility"] = "internal"
+```
+
+## Java
+
+These settings apply only when `--java` is specified on the command line.
+
+``` yaml $(java)
+java: true
```
Comment (jeremymeng) [inline specification/containerregistry/data-plane/Azure.ContainerRegistry/readme.md:56]: @pallavit could you please add your Java customization here as a suggestion comment?

File: specification/containerregistry/data-plane/Azure.ContainerRegistry/readme.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+use-core-v2: true
+use-extension:
+  "@autorest/typescript": "C:/github/autorest.typescript"
+```
+
+## Python
+These settings apply only when `--python` is specified on the command line.
+
+``` yaml $(python)
+python: true
```
Comment (jeremymeng) [inline specification/containerregistry/data-plane/Azure.ContainerRegistry/readme.md:86]: @seankane-msft could you please add your Python customization here as a suggestion comment?

File: specification/containerregistry/data-plane/Azure.ContainerRegistry/readme.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+```
+
+## Suppression
```
Comment (jeremymeng) [inline specification/containerregistry/data-plane/Azure.ContainerRegistry/readme.md:89]: The Suppression section is copied from existing master branch 

https://github.com/Azure/azure-rest-api-specs/blob/master/specification/containerregistry/data-plane/readme.md#suppression

**Review by tjprescott — APPROVED** [review]

**Comment by tjprescott** [comment]
@jeremymeng is this ready to merge?

**Comment by jeremymeng** [comment]
> @jeremymeng is this ready to merge?

@tjprescott I contacted and would like to get some review and approval from the service team first.

File: specification/containerregistry/data-plane/Azure.ContainerRegistry/preview/2019-08-15-preview/containerregistry.json [hunk 5]
```diff
@@ -0,0 +1,3090 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Container Registry",
```
Comment (shahzzzam) [inline specification/containerregistry/data-plane/Azure.ContainerRegistry/preview/2019-08-15-preview/containerregistry.json:4]: Azure Container Registry?

File: specification/containerregistry/data-plane/Azure.ContainerRegistry/preview/2019-08-15-preview/containerregistry.json [hunk 6]
```diff
@@ -0,0 +1,3090 @@
...
+          "x-ms-client-flatten": true
+        }
+      },
+      "example": {
+        "registry": "registryname.azurecr.io",
+        "changeableAttributes": {
+          "readEnabled": true,
+          "listEnabled": true,
+          "deleteEnabled": true,
+          "writeEnabled": true
```
Comment (shahzzzam) [inline specification/containerregistry/data-plane/Azure.ContainerRegistry/preview/2019-08-15-preview/containerregistry.json:1353]: can you add `"teleportEnabled": false`?

File: specification/containerregistry/data-plane/Azure.ContainerRegistry/preview/2019-08-15-preview/containerregistry.json (hunk 5 above)
Comment (jeremymeng) [reply specification/containerregistry/data-plane/Azure.ContainerRegistry/preview/2019-08-15-preview/containerregistry.json]: Updated

File: specification/containerregistry/data-plane/Azure.ContainerRegistry/preview/2019-08-15-preview/containerregistry.json (hunk 6 above)
Comment (jeremymeng) [reply specification/containerregistry/data-plane/Azure.ContainerRegistry/preview/2019-08-15-preview/containerregistry.json]: Added

**Comment by tjprescott** [comment]
@jeremymeng please ping me when you are ready to merge this

**Review by shahzzzam — APPROVED** [review]
LGTM

**Comment by jeremymeng** [comment]
@tjprescott It's ready to go!

**Pull request merged**